util/topgen.py

```diff
@@ -67,7 +67,10 @@ def ipgen_render(template_name: str, topname: str, params: Dict,
 
     Aborts the program execution in case of an error.
     """
-    instance_name = f"top_{topname}_{template_name}"
+    module_name = template_name
+    if "module_instance_name" in params:
+        module_name = params["module_instance_name"]
```

Comment:
Simpler: `module_name = params.get('module_instance_name', template_name)`

Reply:
Thanks for the suggestion. Done.